flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/CheckpointCommittablesImpl.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.operators.sink.committables;
+
+import org.apache.flink.api.connector.sink2.Committer;
+import org.apache.flink.streaming.api.connector.sink2.CommittableSummary;
+import org.apache.flink.streaming.api.connector.sink2.CommittableWithLineage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+class CheckpointCommittablesImpl<CommT> implements CheckpointCommittables<CommT> {
+    private final Map<Integer, SubtaskCommittables<CommT>> subtasksCommittables;
+    private final CommittableCollector<CommT> collector;
+    private final long checkpointId;
+
+    CheckpointCommittablesImpl(CommittableCollector<CommT> collector, long checkpointId) {
+        this.collector = checkNotNull(collector);
+        this.checkpointId = checkpointId;
+        this.subtasksCommittables = new HashMap<>();
+    }
+
+    CheckpointCommittablesImpl(
+            Map<Integer, SubtaskCommittables<CommT>> subtasksCommittables, long checkpointId) {
```

Comment:
I could actually happen that the `subtaskCommittables` have lower checkpoint Ids because in Flink it can happen that a checkpoint subsumes a previous one.

Reply:
good point, I would suggest to add this info as java doc.

Reply:
I added a java doc